sdk/storage/Azure.Storage.Files.DataLake/api/Azure.Storage.Files.DataLake.netstandard2.0.cs

```diff
@@ -146,6 +146,7 @@ public partial class DataLakeFileSystemClient
         public virtual System.Threading.Tasks.Task<Azure.Response<Azure.Storage.Files.DataLake.Models.FileSystemAccessPolicy>> GetAccessPolicyAsync(Azure.Storage.Files.DataLake.Models.DataLakeRequestConditions conditions = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual Azure.Storage.Files.DataLake.DataLakeDirectoryClient GetDirectoryClient(string directoryName) { throw null; }
         public virtual Azure.Storage.Files.DataLake.DataLakeFileClient GetFileClient(string fileName) { throw null; }
+        public virtual Azure.Storage.Files.DataLake.DataLakePathClient GetPathClient(string pathName) { throw null; }
```

Comment:
I'm on the fence about whether we want this.  Or at least to expose it in this fashion.  We intentionally hid the accessor here so customers wouldn't have to understand the class hierarchy when getting started.  If you just gave me `Path`, `Directory`, and `File`, it wouldn't be obvious to me that `Path` was a base class.

What if we just add a .ctor overload to `DataLakePathClient` that takes in a `DataLakeFileSystemClient`?  That would enable this scenario without complicated the on ramp.

Reply:
I believe the customer wants ```DataLakeFileSystemClient.GetPathClient()``` to make it easier to write mocked unit tests.  @khshah3, would you be willing to elaborate on the scenario?

Reply:
Sure - and I think we can achieve the same thing by adding a DataLakePathClient .ctor that behaved the same way but wasn't as enticing to new users.

Reply:
To add context from my original issue:

An access pattern we often run into is that given a path fetch/do something and we don't care about whether its a file/directory. (i.e DataLakePathClient is enough to execute). For example, with DeleteIfExists. It would be invaluable for us to how unit-test friendly way to access the path client. Unit-test friendly is key here as well. 

Given the current access pattern, we are able to mock out GetXClient and therefore it simplifies testability. We can obviously decorate around a ctr for similar functionality. But it seems valuable for it be consistent.

Reply:
As Karan told if we go "new constructor" way then we will need to create a factory or a wrapper. But what is more panful we will need to pass it around everywhere.

Reply:
Assuming you have a `DataLakeFileSystemClient fileSystemClient` and `string path`, I do not understand how `new DataLakePathClient(fileSystemClient, path)` would not work in place of `fileSystemClient.GetPathClient(path)` in a unit test.

I appreciate the desire for consistency, but not exposing `GetPathClient` would be consistent with the broader pattern of not exposing convenience accessors for potentially confusing base types (i.e., `BlobBaseClient` doesn't have a corresponding `BlobContainerClient.GetBlobBaseClient(...)` for this reason).  If we had a `DataLake.Specialized` namespace then I'd be fine with moving the accessor there as an extension method, but I think a .ctor overload is probably the best option given the shape of what has already been GA'ed.

Reply:
Understood. I don't think new DataLakePathClient(fileSystemClient, path) exists. It would be useful to have a constructor which assumes relative path and filesystem client i.e something equivalent to below (abstracted away)

new DataLakePathClient(Uri.AppendToPath(directoryName), Pipeline, Version, ClientDiagnostics);.

This way we can call new DataLakePathClient(fileSystemClient, path)

Reply:
I've updated the review, adding ```DataLakePathClient(fileSystemClient, path)``` and removing ```DataLakeFileSystemClient.GetPathClient(...)```